### ISSUE TITLE
[ISSUE] HSET should be variadic

### ISSUE THREAD

**Issue by Oloompa** [luin/ioredis#1039]
Hi,

since redis V4 HSET is variadic just like hmset (which has been deprecated)

> * HSET is now variadic, and HMSET is considered deprecated (but will be supported for years to come). Please use HSET in new code.
> 
> source: https://raw.githubusercontent.com/antirez/redis/4.0/00-RELEASENOTES

**Comment by Oloompa** [comment]
it's still an issue

**Comment by tuananh** [comment]
is it an error in the type? because i saw this test here which mean it does work

https://github.com/luin/ioredis/commit/51b14786eef4c627c178de4967434e8d4a51ebe0

```js
var redis = new Redis();
        redis.hset("foo", "a", 1, "b", "e", function(err, result) {
          expect(result).to.eql(2);
          redis.hget("foo", "b", function(err, result) {
            expect(result).to.eql("e");
            done();
          });
        });
```

**Comment by Oloompa** [comment]
yes it's in typescript

**Comment by tuananh** [comment]
then it's ought to be fixed in here https://www.npmjs.com/package/@types/ioredis

to bad, the types def is in separated package for now so there are many issues like this

**Comment by Oloompa** [comment]
@types repository's issues is quite a "dump". That's why i published issue there.

As you said types should be managed directly on this repo.

I will open a new ticket for that.

**Comment by Oloompa** [comment]
https://github.com/luin/ioredis/issues/1053#issue-561515519

**Comment by Oloompa** [comment]
useless bot

**Comment by h0tw4t3r** [comment]
Fixed in 51b1478

**Comment by marcbachmann** [comment]
The types of the upcoming version 5 of ioredis got completely refactored.
This is definitely fixed https://luin.github.io/ioredis/classes/default.html#hset
